types/field_type.go

```diff
@@ -231,8 +231,12 @@ func (ft *FieldType) CompactStr() string {
 			suffix = fmt.Sprintf("(%d", displayFlen)
 			if isDecimalNotDefault {
 				suffix += fmt.Sprintf(",%d", displayDecimal)
+			} else {
+				suffix += fmt.Sprintf(",%d", 0)
 			}
 			suffix += ")"
+		} else {
+			suffix = fmt.Sprintf("(%d,0)", defaultFlen)
```

Comment:
It would be simpler if calculate the `displayFlen` and `displayDecimal` first, then just Sprintf once.
```go
suffix = fmt.Sprintf("(%d,%d)", displayFlen, displayDecimal)
```

Reply:
When the type is mysql.TypeNewDecimal, why still need to check isFlenNotDefault or isDecimalNotDefault? What about just using `suffix = fmt.Sprintf("(%d,%d)", displayFlen, displayDecimal)` ?

Reply:
@winkyao , have you checked it with MySQL?

Reply:
I think just use `suffix = fmt.Sprintf("(%d,%d)", displayFlen, displayDecimal)` is okay.